packages/compiler/src/collector/expression.ts

```diff
@@ -1,49 +1,124 @@
       // fontSize={24}
       .when(Node.isNumericLiteral, (num) => {
-        return num.getLiteralValue();
+        return types.staticValue(num.getLiteralValue());
       })
       // fontSize={"24px"}
       .when(Node.isStringLiteral, (str) => {
-        return str.getLiteralValue().trim();
+        return types.staticValue(str.getLiteralValue().trim());
       })
       // fontSize={`24px`}
       .when(Node.isNoSubstitutionTemplateLiteral, (literal) => {
-        return literal.getLiteralValue().trim();
+        return types.staticValue(literal.getLiteralValue().trim());
       })
       // horizontal={true}
       // eslint-disable-next-line @typescript-eslint/unbound-method -- FIXME
       .when(Node.isTrueLiteral, (bool) => {
-        return bool.getLiteralValue();
+        return types.staticValue(bool.getLiteralValue());
       })
       // horizontal={false}
       // eslint-disable-next-line @typescript-eslint/unbound-method -- FIXME
       .when(Node.isFalseLiteral, (bool) => {
-        return bool.getLiteralValue();
+        return types.staticValue(bool.getLiteralValue());
       })
       // fontSize={['24px', '32px']}
       .when(Node.isArrayLiteralExpression, (array) => {
-        const arrayExpression = array.getElements().map((elm) => {
-          return handleJsxExpression(elm);
-        }) as (string | number | undefined)[];
+        const arrayExpression = array
+          .getElements()
+          .map((elm) => handleJsxExpression(elm))
+          .map((x) =>
+            x?.type === "Static"
+              ? typeof x.value === "string" || typeof x.value === "number"
+                ? x.value
+                : undefined
+              : undefined
+          );
         // If the arrayExpression includes undefined, it means some of its elements
         // could not be statically analyzed, hence handle it at runtime.
-        return arrayExpression.includes(undefined)
+        return arrayExpression.some((x) => x === undefined)
           ? undefined
-          : arrayExpression;
+          : types.staticValue(arrayExpression);
       })
-      // fontSize={{xl: '2rem'}['xl']}
+      // fontSize={someFlag ? "24px" : "32px"}
+      .when(Node.isConditionalExpression, (conditional) => {
+        const condition = conditional.getCondition();
+        const whenTrue = handleJsxExpression(conditional.getWhenTrue());
+        const whenFalse = handleJsxExpression(conditional.getWhenFalse());
+
+        const ret =
+          whenTrue?.type === "Static" && whenFalse?.type === "Static"
+            ? types.conditional({
+              expression: condition.getText(),
+              whenTrue: whenTrue.value,
+              whenFalse: whenFalse.value,
+            })
+            : undefined;
+        return ret;
+      })
+      // _hover={{color: 'red'}}
       .when(Node.isObjectLiteralExpression, (obj) => {
-        // TODO
-        return undefined;
+        const entries: [string, types.Value][] = [];
```

Comment:
I mostly copied the logic in `pseudo.ts`, which is now removed.
Note that the original logic had the following bug, which is fixed in my PR:

## Computed properties were silently erased
`_hover` prop was removed from `<Box _hover={{ ['color']: 'red'}} />` without generating the red color CSS. In my PR, computed properties just deopt.